common/rfb/SConnection.cxx

```diff
@@ -524,6 +524,13 @@ void SConnection::authSuccess()
 {
 }
 
+void SConnection::desktopReady()
+{
```

Comment:
This can be called in various states. You need to check that it is the right one before proceeding.